### DISCUSSION TITLE
Is it possible to use Device Label Attribute on Alam Notification?

### DISCUSSION THREAD

**Discussion by maornek** [thingsboard/thingsboard#11544] [General]
I am a satisfied customer of Thingsboard, but I need to display my device label on alarm notifications. But as i couldn't display it neither in web, sms nor mail notifications.

I tried all but none is working.

${alarmOriginatorLabel}
${OriginatorLabel}
${deviceName}
{{deviceLabel}}

As I understand we can't use it on "Alarm" notification, although Device Label attribute can be used in "Device Activity" type of notifications. (https://thingsboard.io/docs/user-guide/notifications/)

We have been using Thingsboard for more than 2 years, and I can clearly suggest allowing users (us) to use "Device Labels" as device aliases to generate notifications for devices with more familiar names for our customers. It would be catastrophic to rename all devices on the TTN side due to creating all new devices on the Thingsboard side and disregarding all history data as we rename devices.

As a customer feedback, I strongly suggest you to allow us to use Device Labels for notifications as well as Device ID/Names, regarding the issues i stated above.

Thanks for your support and understanding all over the way,

May the force be with you all,

Dr. Ornek

**Comment by sprijk** [comment]
I think this could be pretty easy as this information is alread present in `AlarmInfo`
https://github.com/thingsboard/thingsboard/blob/d66a3297b909dc19b8aa5cb64fd69f7eb9594095/common/data/src/main/java/org/thingsboard/server/common/data/alarm/AlarmInfo.java#L39
and add `.alarmOriginatorLabel(alarmInfo.getOriginatorLabel()` under this line
https://github.com/thingsboard/thingsboard/blob/d66a3297b909dc19b8aa5cb64fd69f7eb9594095/application/src/main/java/org/thingsboard/server/service/notification/rule/trigger/AlarmAssignmentTriggerProcessor.java#L64

This way the alarm originator can be designated as i.e. 'Domestic Battery' instead of 'MAC03034' (or to whatever specific ID or serial the device is named).

am I wrong?

I'd love to see this feature as well, thanks in advance!

**Comment by maornek** [comment]
Thanks sprijk, for your great support. As a Community Edition user, I would
like to see that feature as well :) But i am still waiting and trying to
solve problem with cosmetic changes 🤦🏻‍♂️ Help me if anyone can make this
feature happen

On Thu, Nov 7, 2024 at 4:20 PM sprijk ***@***.***> wrote:

> I think this could be pretty easy as this information is alread present in
> AlarmInfo
>
> https://github.com/thingsboard/thingsboard/blob/633411f9e13bbdfb9907dd8ac5bb44507d463937/common/data/src/main/java/org/thingsboard/server/common/data/alarm/AlarmInfo.java#L39
> and add .alarmOriginatorLabel(alarmInfo.getOriginatorLabel() under this
> line
>
> https://github.com/thingsboard/thingsboard/blob/633411f9e13bbdfb9907dd8ac5bb44507d463937/application/src/main/java/org/thingsboard/server/service/notification/rule/trigger/AlarmAssignmentTriggerProcessor.java#L64
>
> This way the alarm originator can be designated as i.e. 'Domestic Battery'
> instead of 'MAC03034' (or to whatever specific ID or serial the device is
> named).
>
> am I wrong?
>
> I'd love to see this feature as well, thanks in advance!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/thingsboard/thingsboard/discussions/11544#discussioncomment-11179192>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABOOJN3JT653EA22636HBM3Z7OAK7AVCNFSM6AAAAABNO5H7EGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJXHEYTSMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 

Dr. Muhammed Ali Örnek

Co-Founder, ForestGuard™

Binary Labs Technology Inc.

www.nowildfire.com

***@***.***

​+33 781 200160

+90 541 3606637

**Comment by ttrowell** [comment]
My workaround has been to add originator attributes in the Rule Chain and gather a Server attribute.

**Reply by getafix6** [reply]
How do you access that in the template?

**Reply by tararowell** [reply]
![image](https://github.com/user-attachments/assets/ca36bd4f-061b-47aa-8a5e-48c936aed87b)

![image](https://github.com/user-attachments/assets/8171a3d9-7256-4166-9ffc-f77cedad1ad9)

![image](https://github.com/user-attachments/assets/7b6e9b34-201b-4b8a-8607-631aad6d1f47)


I hope this helps!
